tidy3d/components/spice/sources/ac.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+from typing import Optional
+
+import pydantic.v1 as pd
+
+from tidy3d.components.base import Tidy3dBaseModel
+from tidy3d.components.types import ArrayFloat1D
+from tidy3d.constants import HERTZ, VOLT
+from tidy3d.constants import inf as td_inf
+
+
+class SSACVoltageSource(Tidy3dBaseModel):
+    """
+    Small-signal AC voltage source.
+
+    This source represents a small-signal AC excitation defined by a list of bias voltages,
+    a set of small-signal analysis frequencies, and a small-signal amplitude (linear scale).
+
+    Notes
+    -----
+    - The bias `voltage` refer to the DC operating point above the simulation ground.
+    - The `frequency` array lists the AC analysis frequencies (in Hz).
+    - The `amplitude` is the small-signal AC magnitude (in Volts).
+    - Currently, electrical ports are not defined.
+
+    Examples
+    --------
+    >>> import tidy3d as td
+    >>> voltages = [-0.5, 0, 1, 2, 3, 4]
+    >>> frequency = [1e3, 1e4, 1e5]
+    >>> amplitude = 1e-3
+    >>> voltage_source = td.SSACVoltageSource(
+    ...     voltage=voltages,
+    ...     frequency=frequency,
+    ...     amplitude=amplitude,
+    ... )
+    """
+
+    name: Optional[str]
+
+    voltage: ArrayFloat1D = pd.Field(
+        ...,
+        title="DC Bias Voltages",
+        description="List of DC operating point voltages (above ground) used with 'VoltageBC'.",
+        units=VOLT,
+    )
+
+    frequency: ArrayFloat1D = pd.Field(
+        ...,
+        title="AC Analysis Frequencies",
+        description="List of small-signal analysis frequencies (in Hz).",
+        units=HERTZ,
+    )
+
+    amplitude: pd.FiniteFloat = pd.Field(
+        ...,
+        title="Small-Signal Amplitude",
+        description="Small-signal AC amplitude (linear scale, in Volts).",
+        units=VOLT,
+    )
+
+    @pd.validator("voltage")
+    def validate_voltage(cls, val):
+        for v in val:
+            if v == td_inf:
+                raise ValueError(f"Voltages must be finite. Current voltage= {val}.")
+        return val
+
+    @pd.validator("frequency")
+    def validate_frequency(cls, val):
+        for v in val:
+            if v == td_inf or v <= 0:
+                raise ValueError(
+                    f"Frequencies must be strictly positive and finite. Current frequency={val}."
+                )
+        return val
+
+    @pd.validator("amplitude")
+    def validate_amplitude(cls, val):
+        if val == td_inf:
+            raise ValueError(f"Small signal amplitude must be finite. Current amplitude={val}.")
```

Comment:
Thinking out loud here, we want the amplitude to not just be finite, but "small" in some sense. Is there something that we can test it against and at least issue a warning if it doesn't seem to be the case?